graylog2-server/src/main/java/org/graylog2/entitygroups/model/EntityGroup.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.entitygroups.model;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import com.google.common.graph.MutableGraph;
+import jakarta.annotation.Nullable;
+import org.graylog.autovalue.WithBeanGetter;
+import org.graylog2.contentpacks.EntityDescriptorIds;
+import org.graylog2.contentpacks.model.entities.EntityDescriptor;
+import org.graylog2.entitygroups.contentpacks.entities.EntityGroupEntity;
+import org.graylog2.contentpacks.ContentPackable;
+import org.graylog2.database.entities.ScopedEntity;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@AutoValue
+@JsonAutoDetect
+@JsonDeserialize(builder = EntityGroup.Builder.class)
+@WithBeanGetter
+public abstract class EntityGroup extends ScopedEntity implements ContentPackable<EntityGroupEntity> {
+    public static final String FIELD_NAME = "name";
+    public static final String FIELD_ENTITIES = "entities";
+
+    @JsonProperty(FIELD_NAME)
+    public abstract String name();
+
+    @Nullable
```

Comment:
As noted in the comment for `DBEntityGroupService`, please reconsider whether this should be nullable or default to an empty set.